docs/tutorials/monitoring/index.md

```diff
@@ -0,0 +1,81 @@
+# Monitoring & Performance
+
+A Feldera instance and its pipelines can be monitored using various tools. This tutorial
+will guide you through setting up monitoring for your Feldera instance.
+
+## Metrics with Grafana and Prometheus
+
+Metrics are helpful to check the health of your Feldera instance and to identify resource
+bottlenecks. Feldera exposes a metrics endpoint that can be scraped by Prometheus. Grafana
+is then used to visualize these metrics.
+
+### Setup
+
+1. **Prometheus:** You must have [Prometheus installed](https://prometheus.io).
+2. **Connect Prometheus to Feldera:**
+    - Add the following to your `prometheus.yml` configuration file, usually located in `/etc/prometheus/prometheus.yml`:
+    ```yaml
+    - job_name: 'feldera'
+      scrape_interval: 1s
+      static_configs:
+        - targets: ['localhost:8081']
+    ```
+    - Replace `localhost:8081` with the address of your Feldera instance. Note that `8081` is the default port for metrics.
+    - Restart Prometheus.
+3. **Grafana:** You must have [Grafana installed](https://grafana.com).
+4. **Add Prometheus To Grafana:**
+    - If you are using a local prometheus instance, the URL for the Prometheus data source will be `http://localhost:9090`.
+    - Follow the steps in the [Grafana
+      documentation](https://prometheus.io/docs/visualization/grafana/#creating-a-prometheus-data-source)
+      to add Prometheus as a data source.
+ 
+### Set up the monitoring Dashboard
+
+1. **Copy the JSON of the [Feldera template dashboard](https://raw.githubusercontent.com/feldera/feldera/main/deploy/grafana_dashboard.json)**
+2. **Import the dashboard into Grafana**
+   - Under Dashboards, click the "New" icon, then click "Import Dashboard".
+   - Paste the JSON copied from the template in the text-box and click "Load".
+
+A Feldera Instance Monitoring dashboard will be created in Grafana.
+The dashboard is a template and may need to be adjusted to fit your specific needs.
+Look for the `feldera_*` metrics in Grafana to add more metrics to the dashboard.
+
+## Tracing with Jaeger
+
+A Feldera pipeline can be traced using Jaeger. Tracing is useful if you need to analyze
+throughput or latency bottlenecks of your pipeline, as it instruments every step of the
+pipeline execution and provides detailed information about the execution time of each
+step.
+
+### Setup
+
+1. **Jaeger:** You must have [Jaeger installed](https://www.jaegertracing.io).
+2. Start Jaeger using the `jaeger-all-in-one` script:
+    ```bash
+    ./jaeger-all-in-one
+    ```
+3. **Enable tracing in a Feldera pipeline**
+   - If you're running both a pipeline and jaeger on the same machine, by default, the pipeline
```

Comment:
Tracing is on by default? Or do you mean if you only enable tracing, it will find the jaeger instance this on the same machine?

Btw, it looks like these instructions/defaults are tuned to only work with `cargo run`. They won't work for Docker etc.

Reply:
clarified the docs. I am too stupid to understand docker, but I guess the right approach would be to put all these services in the docker-compose and wire them up correctly

Reply:
Yup. :)